File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Views/ViewUser.cs
using KnowledgeSharingApi.Domains.Models.Entities.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Views
{
    [Table("ViewUser")]
    public class ViewUser : Profile
    {
        public string Email {  get; set; } = string.Empty;
        public string Username { get; set; } = string.Empty;
        public string Role { get; set; } = string.Empty;

        public int? TotalFriend { get; set; }
        public int? TotalRequester { get; set; }
        public int? TotalRequestee { get; set; }
        public int? TotalFollower { get; set; }
        public int? TotalFolowee { get; set; }
        public int? TotalBlocker { get; set; }
        public int? TotalBlockee { get; set; }

        protected override ViewUser Init()
        {
            return new ViewUser();
        }
    }
}

File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/views/view-user.js
// Framework and Technology Stack: C#

import Profile from "../entities/profile";

class ViewUser extends Profile {
    constructor() {
        super();
        this.Email = null;
        this.Username = null;
        this.Role = null;

        this.TotalFriend = null; 
        this.TotalRequester = null; 
        this.TotalRequestee = null; 
        this.TotalFollower = null; 
        this.TotalFolowee = null; 
        this.TotalBlocker = null; 
        this.TotalBlockee = null; 
    }

    init() {
        return new ViewUser();
    }
}

export default ViewUser;

